com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -395,6 +400,47 @@ class Job(_ResourceElement):
         >>> print (jobs[0].health)
         healthy
     """
+    def get_application_trace(self, path=None, name=None):
+        """Retrieves the application log and saves it to the specified path with the given name as a tar file.
+
+        If logs are retrieved with the same path and name as previously retrieved logs, the prior logs will be
+        overwritten.
+
+        Attributes:
+            path (str): a valid directory in which to save the application log output. Defaults to current dir.
+            name (str): the filename of the created tar file. Defaults to a name based on the job name.
+
+         Returns:
+            str: the path to the application logs tar file.
+         """
+        logger.debug("Retrieving application logs from: " + self.applicationLogTrace)
+        logs = self.rest_client.make_raw_request(self.applicationLogTrace)
+        
+        if name is None:
+            # Take the job name and remove colons (colons confuse the unix 'tar' command)
+            name = ''.join(self.name.split(':'))
```

Comment:
Can you include the job id, so that multiple runs of a job with the same name produce different files?

Reply:
Added to the name.